### ISSUE TITLE
Add multiple legends to a single ggpairs

### ISSUE THREAD

**Issue by pshukla63** [ggobi/ggally#486]
Hello! I have a rather complex ggpairs plot that I am trying to plot. I want the color gradient legend and fill legend to both appear but I don't know how to achieve this. This is example code: 

```
lowerFn <- function(data, mapping) {
  ggplot(data = data, mapping = mapping) +
    geom_point(alpha = 0.6) +
    scale_color_gradient2(low = "#B2182B", midpoint = 1, high = "#2166AC")
}

upperFn <- function(data, mapping) {
  ggplot(data = data, mapping = mapping) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = my_colors)
}

my_colors <- c(A = "green", B = "purple", C = "yellow")

df <- data.frame(PC1 = rnorm(100), PC2 = rnorm(100), PC3 = rnorm(100), 
                 type = sample(c("A", "B", "C"), 100, replace = TRUE), distance = rnorm(100))


ggpairs(df %>% select(starts_with("PC")),
  upper = list(
    mapping = aes(color = df$type),
    continuous = upperFn
  ),
  lower = list(
    mapping = aes(color = df$distance),
    continuous = lowerFn
  ),
  diag = list(mapping = aes(fill = df$type)), legend = 1
) +
  theme_bw() + scale_fill_manual(values = my_colors) + labs(fill = "Type")

```

Thank you!

**Comment by touss643** [comment]
Hello. I am new to the package and together with @Ange-G-K and
under the supervision of @JMeyer31 we are willing to solve this
issue, as part of an assignment at the ENSAI school of statistical engineering
(France). We will try to be as autonomous as possible but we may ask for guidance from now and then. We
have a solid background in statistics and a reasonable one in computer science.
If any maintainer thinks this issue is too ambitious for a beginner, please tell
us so and - if you have time - guide us to a project / issue were our help may
be more useful.

**Comment by katossky** [comment]
Let me rephrase : you would like `ggpairs` to correctly handle the corner-cases where there may be more than one guide ?

@touss643  @Ange-G-K, you may want to try to see if the behaviour is the same for other aesthetics, like point shapes or line type. In that case it is probably a generic behaviour of the guide.

**Comment by schloerke** [comment]
While a little late, this was a tricky one!

``` r
library(GGally)
#> Loading required package: ggplot2
#> Registered S3 method overwritten by 'GGally':
#>   method from   
#>   +.gg   ggplot2
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

lowerFn <- function(data, mapping) {
  ggplot(data = data, mapping = mapping) +
    geom_point(alpha = 0.6) +
    scale_color_gradient2(low = "#B2182B", midpoint = 1, high = "#2166AC")
}

upperFn <- function(data, mapping) {
  ggplot(data = data, mapping = mapping) +
    geom_point(alpha = 0.5) +
    scale_color_manual(values = my_colors)
}

my_colors <- c(A = "green", B = "purple", C = "yellow")

df <- data.frame(PC1 = rnorm(100), PC2 = rnorm(100), PC3 = rnorm(100), 
                 type = sample(c("A", "B", "C"), 100, replace = TRUE), distance = rnorm(100))

make_pm <- function(legend = NULL) {
  ggpairs(
    df %>% select(starts_with("PC")),
    upper = list(
      mapping = aes(color = df$type),
      continuous = upperFn
    ),
    lower = list(
      mapping = aes(color = df$distance),
      continuous = lowerFn
    ),
    diag = list(mapping = aes(fill = df$type)),
    legend = legend
  ) +
    theme_bw() + 
    scale_fill_manual(values = my_colors) + 
    labs(fill = "Type")
}

pm_init <- make_pm()

gradient_legend <- grab_legend(pm_init[2,1] + labs(color = "Distance"))
#> Warning: No shared levels found between `names(values)` of the manual scale and the
#> data's fill values.
fill_legend <- grab_legend(pm_init[1,1])

stacked_legends <- gtable::gtable_col(
  "guide-box-right",
  list(
    gtable::gtable_col(
      "legends",
      list(
        gradient_legend, 
        fill_legend
      )
    )
  )
)
class(stacked_legends) <- c("legend_guide_box", class(stacked_legends))

pm <- make_pm(legend = stacked_legends)
pm
#> Warning: No shared levels found between `names(values)` of the manual scale and the
#> data's fill values.
#> No shared levels found between `names(values)` of the manual scale and the
#> data's fill values.
#> No shared levels found between `names(values)` of the manual scale and the
#> data's fill values.
#> No shared levels found between `names(values)` of the manual scale and the
#> data's fill values.
#> No shared levels found between `names(values)` of the manual scale and the
#> data's fill values.
#> No shared levels found between `names(values)` of the manual scale and the
#> data's fill values.
```

![](https://i.imgur.com/Uv0hfcs.png)

<sup>Created on 2025-06-11 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
